### DISCUSSION TITLE
What resource sets the sourceDateGutter change color?

### DISCUSSION THREAD

**Discussion by themountinman** [codefori/vscode-ibmi#3038] [Q&A] [answered]
👉🏻 I don't have a bug to report. I just need information, and haven't been able to find it anywhere else.
I would like to know what resource is used to change the color of the source date when you edit the associated line in the source code, and where that resource is stored (e.g. settings.json?).
I'm one of those pain-in-the-rear devs that likes to set all my editor colors myself. And I find the source date change color too attention-grabbing. I would like to change it something more subtle. But, the <i>Developer:  Inspect Editor Tokens and Scopes</i> tool does not seem to work in the line/date gutter, so I don't know how to identify the resource myself.

Any assistance is appreciated.

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.14.5|
|Visual Studio Code version|1.96.0|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
.NET Install Tool (vscode-dotnet-runtime): 2.2.3
COBOL (cobol): 24.12.18
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.5.0
Configuration Editing (configuration-editing): 1.0.0
Db2 for IBM i (vscode-db2i): 1.7.0
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.22.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
IBM i Debug (ibmidebug): 2.0.2
IBM i Project Explorer (vscode-ibmi-projectexplorer): 2.12.0
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Node Debug Auto-attach (debug-auto-launch): 1.0.0
RPGLE (vscode-rpgle): 0.28.5
Source Orbit (vscode-sourceorbit): 1.0.2
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R4M0|
|Tech Refresh|10|
|CCSID Origin|37|
|Runtime CCSID|37|
|Default CCSID|37|
|SSHD CCSID|?|
|cqsh|true|
|SQL|Enabled
|Source dates|Enabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|/QOpenSys/QIBM/ProdData/JavaVM/jdk80|
|-|-|-|-|-|-|
|bash|attr|GETNEWLIBL.PGM|QZDFMDB2.PGM|startDebugService.sh|64bit|
|chsh|iconv|||||
|ls|setccsid|||||
|md5sum|tar|||||
|sort|uname|||||
|stat||||||
|tn5250||||||
</details>

<details>
<summary>Shell env</summary>

```bash
BUILDLIB=<redacted>
CURLIB=<redacted>
HOME=/home/<redacted>
HOST=<redacted>
LIBLS=<recacted>
LOGIN=<redacted>
LOGNAME=<redacted>
MAIL=/var/spool/mail/<redacted>
PASE_USRGRP_LIMITED=N
PATH=/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
SHELL=/QOpenSys/usr/bin/bsh
SSH_CLIENT=<redacted>
SSH_CONNECTION=<redacted>
TZ=<PST>8<PDT>,M3.2.0,M11.1.0
USER=<redacted>
USERNAME=<redacted>
WORKDIR=/home/<redacted>
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$",
  "qsysNameRegex": {}
}
```
</details>

**Comment by SanjulaGanepola** [comment, marked as answer]
@themountinman There are 2 types of gutter colors which have associated `ThemeColor` IDs:
- For searched dates: `gitDecoration.addedResourceForeground`
- For lines edited today: `gitDecoration.modifiedResourceForeground`

These are default IDs part of VS Code which can in fact be customized. You can do this in the VS Code settings:

<img width="1657" height="723" alt="Image" src="https://github.com/user-attachments/assets/53e1cf81-9cad-4dcf-937d-432a4690ddb5" />

and then customizing the color associated with those IDs:

<img width="2243" height="515" alt="Image" src="https://github.com/user-attachments/assets/005f9ce8-533a-449c-b92a-6749392d5569" />

You can find all these different built-in `ThemeColor` IDs here: https://code.visualstudio.com/api/references/theme-color
